sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultAccessControlAsyncClient.java

```diff
@@ -185,32 +204,311 @@ HttpPipeline getHttpPipeline() {
+        return withContext(context -> getRoleDefinitionWithResponse(roleScope, roleDefinitionName, context));
+    }
+
+    /**
+     * Gets a {@link KeyVaultRoleDefinition}.
+     *
+     * @param roleScope The {@link KeyVaultRoleScope role scope} of the {@link KeyVaultRoleDefinition}.
+     * @param roleDefinitionName The name of the {@link KeyVaultRoleDefinition}.
+     * @param context Additional context that is passed through the HTTP pipeline during the service call.
+     *
+     * @return A {@link Mono} containing a {@link Response} whose {@link Response#getValue() value} contains the
+     * {@link KeyVaultRoleDefinition}.
+     *
+     * @throws NullPointerException If the {@link KeyVaultRoleScope role scope} or {@link String roleDefinitionName} are
+     * {@code null}.
+     */
+    Mono<Response<KeyVaultRoleDefinition>> getRoleDefinitionWithResponse(KeyVaultRoleScope roleScope,
+                                                                         String roleDefinitionName, Context context) {
+        Objects.requireNonNull(roleScope,
+            String.format(KeyVaultErrorCodeStrings.getErrorString(KeyVaultErrorCodeStrings.PARAMETER_REQUIRED),
+                "'roleScope'"));
+        Objects.requireNonNull(roleDefinitionName,
+            String.format(KeyVaultErrorCodeStrings.getErrorString(KeyVaultErrorCodeStrings.PARAMETER_REQUIRED),
+                "'roleDefinitionName'"));
+
+        try {
+            return clientImpl.getRoleDefinitions()
+                .getWithResponseAsync(vaultUrl, roleScope.toString(), roleDefinitionName,
+                    context.addData(AZ_TRACING_NAMESPACE_KEY, KEYVAULT_TRACING_NAMESPACE_VALUE))
+                .doOnRequest(ignored -> logger.verbose("Retrieving role assignment - {}", roleDefinitionName))
+                .doOnSuccess(response ->
+                    logger.verbose("Retrieved role assignment - {}", response.getValue().getName()))
+                .doOnError(error ->
+                    logger.warning("Failed to retrieved role assignment - {}", roleDefinitionName, error))
+                .map(KeyVaultAccessControlAsyncClient::transformRoleDefinitionResponse);
+        } catch (RuntimeException e) {
+            return monoError(logger, e);
+        }
+    }
+
+    /**
+     * Deletes a {@link KeyVaultRoleDefinition}.
+     *
+     * @param roleScope The {@link KeyVaultRoleScope role scope} of the {@link KeyVaultRoleDefinition}. Managed HSM
+     * only supports '/'.
+     * @param roleDefinitionName The name of the {@link KeyVaultRoleDefinition}.
+     *
+     * @return A {@link Mono} containing the {@link KeyVaultRoleDefinition}.
+     *
+     * @throws NullPointerException If the {@link KeyVaultRoleScope role scope} or {@link String roleDefinitionName} are
+     * {@code null}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<KeyVaultRoleDefinition> deleteRoleDefinition(KeyVaultRoleScope roleScope, String roleDefinitionName) {
```

Comment:
What happens when the roleDefinition is not found? Will the delete succeed or will it throw ResourceNotFoundException? Please document the behavior in javadocs.

Reply:
I'm not sure, I'll run some tests to find out.

Reply:
Presumably it'll throw, but it does raise a good point: all these methods should probably document whatever general request exception you throw for service errors. We do in .NET anyway (perhaps not consistently across libraries, but we should).

Reply:
The method does throw if the resource is not found.